Common/Sources/Extensions/String.swift

```diff
@@ -79,6 +79,30 @@ public extension String {
         }
     }
 
+    /// Format an amount to a currency format with currency code
+    ///
+    /// - Parameter currencyCode: the currency code
+    /// - Returns: a formatted String amount with currency code
+    func formatWithActualDecimal(with currencyCode: String?) -> String {
+        if !self.isEmpty {
+            let currencyFormatter = NumberFormatter()
+            let locale = Locale(identifier: Locale.preferredLanguages[0])
+            let localizedDecimalSeparator: Character = locale.decimalSeparator?.first ?? "."
+            let components = self.split(separator: localizedDecimalSeparator)
+            let fractionDigits = components.count <= 1 ? 2 : components[1].count
+            currencyFormatter.maximumFractionDigits = fractionDigits
```

Comment:
Please move the currency formatter logic to a seperate `func`

Reply:
Now following code has been removed.